### PR TITLE
add a theme class to the body

### PR DESCRIPTION
**Pull request by mathieudutour**
@robintindale does that look ok?

### CODE REVIEW

**Comment by robintindale** [comment]
yeah looks good to merge

I will drop a few stylistic comments - but they're pretty much personal taste so feel free to ignore and land

File: lib/set-delegates.js [hunk 1]
```diff
@@ -206,6 +227,24 @@ module.exports = function(browserWindow, panel, webview, options) {
...
 
+  var themeObserver = ThemeObserverClass.new({
+    utils: {
+      executeJavaScript: browserWindow.webContents.executeJavaScript.bind(
+        browserWindow.webContents
+      ),
+    },
+  })
+
+  NSApplication.sharedApplication().addObserver_forKeyPath_options_context(
```
Comment (robintindale) [inline lib/set-delegates.js]: in our implementation, I created `register` and `deregister`  methods on the `ThemeObserverClass` to tie up these `addObserver` and `removeObserver` calls - feels neater for if the keypath was to change or something

File: lib/set-delegates.js [hunk 2]
```diff
@@ -206,6 +227,24 @@ module.exports = function(browserWindow, panel, webview, options) {
     })
   }
 
+  var themeObserver = ThemeObserverClass.new({
+    utils: {
+      executeJavaScript: browserWindow.webContents.executeJavaScript.bind(
+        browserWindow.webContents
```
Comment (robintindale) [inline lib/set-delegates.js]: is it possible to simply pass the `browserWindow` and use `call`? I don't remember exactly how `ObjCClass` handles construction / closures. I do remember this `utils` paradigm from other places in the code-base though, so I'm probably just wishful thinking :)

**Comment by robintindale** [comment]
oh - I didn't test functionality btw. Would you like me to pull and test, or is a code pass OK?

File: lib/set-delegates.js (hunk 2 above)
Comment (mathieudutour) [reply lib/set-delegates.js]: so `browserWindow` is a javascript object while `utils` will be an NSDictionary. I remember some cases where passing a javascript object directly broke Sketch but I can't remember the details now.
I just reuse the same pattern everywhere now hahaha

**Comment by mathieudutour** [comment]
I tested it already, seems to work fine. Thanks!

File: lib/set-delegates.js (hunk 2 above)
Comment (robintindale) [reply lib/set-delegates.js]: ah yeah - makes sense :)

**Pull request merged**